azure/services/dns/nsrecords.go

```diff
@@ -3,69 +3,45 @@ package dns
 import (
 	"context"
 
-	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2018-05-01/dns"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/dns/armdns"
 	"github.com/Azure/go-autorest/autorest/to"
-	"github.com/giantswarm/microerror"
-
-	"github.com/giantswarm/dns-operator-azure/azure"
 )
 
-func (s *Service) createNSRecords(ctx context.Context, nsRecords []azure.NSRecordSetSpec) error {
-	var err error
-	dnsSpec := s.Scope.DNSSpec()
-	if len(nsRecords) == 0 {
-		s.Scope.V(2).Info(
-			"All DNS NS records have already been created",
-			"DNSZone", dnsSpec.ZoneName)
-		return nil
+func (s *Service) deleteClusterNSRecords(ctx context.Context) error {
+
+	err := s.azureBaseZoneClient.DeleteRecordSet(
+		ctx,
+		s.scope.BaseDomainResourceGroup(),
+		s.scope.BaseDomain(),
+		armdns.RecordTypeNS,
+		s.scope.ClusterName(),
+	)
+	if err != nil {
+		return err
 	}
 
-	for _, nsRecord := range nsRecords {
-		var allNSDomainNames string
-		var nsRecords []dns.NsRecord
-		for i, nsdn := range nsRecord.NSDomainNames {
-			if i > 0 {
-				allNSDomainNames += ", " + nsdn.NSDomainName
-			} else {
-				allNSDomainNames += nsdn.NSDomainName
-			}
-
-			nsDomainName := nsdn.NSDomainName
-			nsRecord := dns.NsRecord{
-				Nsdname: &nsDomainName,
-			}
-			nsRecords = append(nsRecords, nsRecord)
-		}
-
-		s.Scope.V(2).Info(
-			"Creating DNS NS record",
-			"DNSZone", dnsSpec.ZoneName,
-			"name", nsRecord.Name,
-			"NSDomainNames", allNSDomainNames)
+	return nil
+}
 
-		recordSet := dns.RecordSet{
-			Type: to.StringPtr(string(dns.NS)),
-			RecordSetProperties: &dns.RecordSetProperties{
-				NsRecords: &nsRecords,
-				TTL:       to.Int64Ptr(nsRecord.TTL),
+// createClusterNSRecord create a NS record in the basedomain
+// for zone delegation
+func (s *Service) createClusterNSRecord(ctx context.Context, nameServerRecords []*armdns.NsRecord) error {
+
+	_, err := s.azureBaseZoneClient.CreateOrUpdateRecordSet(
+		ctx,
+		s.scope.BaseDomainResourceGroup(),
+		s.scope.BaseDomain(),
+		armdns.RecordTypeNS,
+		s.scope.ClusterName(),
+		armdns.RecordSet{
+			Properties: &armdns.RecordSetProperties{
+				TTL:       to.Int64Ptr(3600),
```

Comment:
we are hardcoding TTLs to 3600 ( 1 h ) in multiple places like the `api` records and the `NS record` 

is `1h` the correct value ? 

If we recreate a cluster you will need to invalidate caches or things won't work for up to 1h potentially. 

I think this should be either configurable or shorter

Reply:
I've chosen `1h` as `CAPZ` create the `api` record for private clusters also with this TTL. If you create a new child-Zone (zone-delegation) via Azure, the `NS-record` will also get a TTL of `1h` per default.

I'm totally fine for using a TTL of `30m` as your comment are totally valid.
But i wont propose introducing a new config flag for this. Let's `hardcode` it for now and if we really have the need, we could still make it configurable. WDYT?

Reply:
yeah cool. I was thinking more like using an annotation 

If we don't recreate clusters with same name this should not be an issue and i guess this situation only applies to us testing ( mario01 , fctest1 , ... ? )

Reply:
Maybe let's add it to the README as well ?

Reply:
done